Remove GPU dependency from ProfileObserver

mdschatz opened the pull request:
Summary: Remove GPU dependency and register ProfileObserver.

Differential Revision: D14265801

ezyang commented:
This isn't just removing the dependency, this is removing CUDA functionality completely! Maybe that's the right thing to do (I am not familiar with ProfileObserver), but let the commit message reflect the contents of the diff.

ezyang commented:
It looks like you're the one who added this in the first place, so I'll LGTM it :)

ezyang commented:
@pytorchbot rebase this please

mdschatz commented:
Yep, I'm fine removing CUDA functionality.  It's creating some headaches and we've only really used this for CPU profiling anyway.

pjh5 commented on caffe2/observers/CMakeLists.txt:
```diff
@@ -1,13 +1,10 @@
   )
-  set(Caffe2_CONTRIB_OBSERVERS_GPU_SRC
-    "${CMAKE_CURRENT_SOURCE_DIR}/profile_observer_gpu.cc"
-  )
```
You can delete https://github.com/pytorch/pytorch/blob/master/caffe2/observers/CMakeLists.txt#L15-L16 as well

pjh5 reviewed (commented):
I think you need to add CAFFE2_API to https://github.com/pytorch/pytorch/blob/master/caffe2/observers/profile_observer.h#L99 just like https://github.com/pytorch/pytorch/blob/master/caffe2/observers/time_observer.h#L48 and also to ProfileOperatorObserver etc

Commit:
Remove GPU dependency from ProfileObserver (#17592)

Summary:
Remove GPU dependency and register ProfileObserver.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/17592

Reviewed By: ezyang

Differential Revision: D14265801

Pulled By: mdschatz

fbshipit-source-id: 6f8bf4cf17d1ab50b86d59c29176da644d9e1ebd
